Add documentation about Rock-on restore (#406)
rockstor/rockstor-doc: data_loss.rst

.. code-block:: console

    btrfs balance status /mnt2/pool-name

.. _rockonrestore:

Rock-on Restore
^^^^^^^^^^^^^^^

> When Rock-ons are not visible after data restore it is possible that they are either running silently in the background

Reviewer: ```suggestion
When Rock-ons are not visible after data restore, or pool import, it is possible that they are either running silently in the background
```

Just to emphasise that sometimes we are just re-installing and importing the pool. Which should then be followed by a config restore, but there might not be one and as you intimate if the Rock-ons-root has been re-activated we could have ghost Rock-ons, or at least ghost docker images running after a by-hand rock-ons-root re-config as-was.

or Rockstor does not have the needed information to restore the Rock-ons. Here is how to get your Rock-ons back up and
running.

> Rock-on data was not stored on system drive (recommended)

Reviewer: ```suggestion
Rock-on associated shares are not stored on the system drive (recommended)
```

This way we keep with the Web-UI reference of shares, and cover both the rock-ons-root and the specific Rock-on config/data shares encouraged for each Rock-on.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

1. Restore your data, pools and shares to make sure the Rock-on data is available
2. Restore your Rockstor configuration

This should be enough for Rockstor to restore your Rock-ons and make them visible and manageable from the UI again.

> Rock-on data stored on system drive (discouraged)

Reviewer: ```suggestion
Rock-on associated shares were stored on the system drive (discouraged)
```

Again with the in-Web-UI references. Data is too broad and often used in our Rock-ons docs to reference the persistent volumes more than the Rock-ons-root. But both must be present for function obviously.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> 1. Restore your Rock-on data and the needed shares of your system drive to make sure the Rock-on data is available again (this would be the ideal moment to move the data to a share that is not on the system drive!)

Reviewer: ```suggestion
1. Restore your Rock-on associated shares on your system drive to make sure they are available again (this would be the ideal moment to instead restore/move them to a share that is not on the system drive!)
```

Reword in accordance with the title changes earlier.

> 2. Re-install the Rock-ons with the same configuration you installed them with before. In case you moved the Rock-on data to a share not on the system drive, be sure to point the Rock-on's storage location to the correct share.

Reviewer: Nice.

> 3. Start the Rock-on

Author: Did not know that. Nice.
